lib/ui/painting.dart

```diff
@@ -323,13 +423,21 @@ class Color {
     if (other.runtimeType != runtimeType) {
       return false;
     }
-    return other is Color
-        && other.value == value;
+    // TODO(gaaclarke): Make equality more fine grain than 1/256.
+    return other is Color &&
+        _floatToInt8(other.a) == _floatToInt8(a) &&
+        _floatToInt8(other.r) == _floatToInt8(r) &&
+        _floatToInt8(other.g) == _floatToInt8(g) &&
+        _floatToInt8(other.b) == _floatToInt8(b) &&
+        other.colorSpace == colorSpace;
   }
 
+  // TODO(gaaclarke): Calculate this from float values.
   @override
-  int get hashCode => value.hashCode;
+  int get hashCode => Object.hash(_floatToInt8(a), _floatToInt8(r),
```

Comment:
You should change the hash code computation in this PR and not as a follow up.

Reply:
Responded to this in https://github.com/flutter/engine/pull/54415#discussion_r1717512260